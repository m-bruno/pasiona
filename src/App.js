import { useState } from "react";
import Task from "./components/Task";
import AddTask from "./components/AddTask";

function App() {
  const [tasks, setTasks] = useState([]);

  // Add Task
  const addTask = (task) => {
    const id = Math.floor(Math.random() * 10000) + 1;
    const newTask = { id, ...task };
    setTasks([...tasks, newTask]);
  };

  // Delete Task
  const deleteTask = (id) => {
    const deletedTasks = tasks.filter((task) => task.id !== id);
    setTasks(deletedTasks);
  };

  return (
    <>
      <h1 data-testid="app-title">Pasiona's To Do List</h1>
      <div className="container">
        <div className="row">
          <div className="twelve column">
            <AddTask onAdd={addTask} />
          </div>
          <div className="twelve column">
            <h2>My Tasks</h2>
            {tasks.map((task) => (
              <Task key={task.id} task={task} onDelete={deleteTask} />
            ))}
          </div>
        </div>
      </div>
    </>
  );
}

export default App;
