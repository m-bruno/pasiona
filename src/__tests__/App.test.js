import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import userEvent from "@testing-library/user-event";
import App from "../App";

test("<App /> Rendering App", () => {
  render(<App />);

  //App title.
  const appTitle = screen.getByTestId("app-title");
  expect(appTitle).toBeInTheDocument();
  expect(appTitle.tagName).toBe("H1");
  expect(appTitle.tagName).not.toBe("H2");
  expect(appTitle.textContent).toBe("Pasiona's To Do List");

  // New Task title.
  expect(screen.getByText("New Task")).toBeInTheDocument();

  // My Tasks title.
  expect(screen.getByText("My Tasks")).toBeInTheDocument();
});

test("<App /> Testing saved tasks exist in the DOM", async () => {
  render(<App />);

  // Filling in input and clicking on Save Task button.
  const taskInput = screen.getByTestId("task-input");
  userEvent.type(taskInput, "Buy groceries");
  const btnSubmit = screen.getByTestId("btn-submit");
  userEvent.click(btnSubmit);

  // Testing that alert isn't displaying when input was correctly filled in.
  const alert = screen.queryByTestId("alert");
  expect(alert).not.toBeInTheDocument();

  // Creating snapshot to test saved tasks exist in the DOM.
  const tasks = await screen.findAllByTestId("savedtask");
  expect(tasks).toMatchSnapshot();

  // Testing specific saved task exists in the DOM.
  expect(screen.getByText("Buy groceries")).toBeInTheDocument();
});

test("<App /> Testing delete button works", () => {
  render(<App />);

  // Filling in input and clicking on Save Task button.
  const taskInput = screen.getByTestId("task-input");
  userEvent.type(taskInput, "Buy groceries");
  const btnSubmit = screen.getByTestId("btn-submit");
  userEvent.click(btnSubmit);

  // Testing that alert isn't displaying when input was correctly filled in.
  const alert = screen.queryByTestId("alert");
  expect(alert).not.toBeInTheDocument();

  // Testing delete button exists.
  const deleteButton = screen.getByTestId("btn-eliminar");
  expect(deleteButton.tagName).toBe("BUTTON");
  expect(deleteButton).toBeInTheDocument();

  // Testing delete button clicks.
  userEvent.click(deleteButton);

  // Testing delete button doesn't exist anymore after deleting task.
  expect(deleteButton).not.toBeInTheDocument();

  // Testing specific saved task doesn't exist anymore after deleting.
  expect(screen.queryByText("Buy groceries")).not.toBeInTheDocument();
});
