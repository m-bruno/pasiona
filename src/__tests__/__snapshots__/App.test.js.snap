// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Testing saved tasks exist in the DOM 1`] = `
Array [
  <div
    class="task-container"
    data-testid="savedtask"
  >
    <h3>
      Buy groceries
    </h3>
    <button
      class="delete"
      data-testid="btn-eliminar"
    >
      X
    </button>
  </div>,
]
`;
